src/ploomber/tasks/notebook.py

```diff
@@ -757,7 +784,18 @@ def _init_source(source,
                      ext_in=None,
                      kernelspec_name=None,
                      static_analysis='regular',
-                     check_if_kernel_installed=False):
+                     check_if_kernel_installed=False,
+                     extract_up=False,
+                     extract_prod=False):
+        ns = NotebookSource(
+            source,
+            ext_in=ext_in,
+            kernelspec_name=kernelspec_name,
+            static_analysis=static_analysis,
+            check_if_kernel_installed=check_if_kernel_installed,
+            **kwargs)
+        NotebookMixin._validate_parameters_cell(ns, extract_up, extract_prod)
+        # re-create NotebookSource instance
         return NotebookSource(
```

Comment:
do we need to initialize this two times? why not

```
ns  = NotebookSource(...)
ns._validate_parameters_cell(extract_up, extract_prod)
return ns
```

Reply:
Working on it